godot-core/src/builtin/dictionary.rs

```diff
@@ -320,6 +331,202 @@ impl Share for Dictionary {
     }
 }
 
+struct DictionaryIterator<'a> {
+    next_var: Option<Variant>,
+    dictionary: &'a Dictionary,
+}
+
+impl<'a> DictionaryIterator<'a> {
+    fn call_init(dictionary: &Dictionary) -> Option<Variant> {
+        let dictionary = dictionary.to_variant();
+        let variant: Variant = Variant::nil();
+        let mut valid = false;
+        // SAFETY:
+        // `dictionary` is a valid `Dictionary` since we have a reference to it,
+        //    so this will call the implementation for dictionaries.
+        // `variant` is an initialized and valid `Variant`.
+        // `variant_iter_init` will set `valid` to a valid `bool` value.
+        let success = unsafe {
+            interface_fn!(variant_iter_init)(
+                dictionary.var_sys(),
+                variant.var_sys(),
+                &mut valid as *mut bool as *mut u8,
+            )
+        };
+        // The function returns a bool
+        assert!(success == 0 || success == 1);
+        let success = success != 0;
+
+        if success {
+            assert!(valid);
+            Some(variant)
+        } else {
+            None
+        }
```

Comment:
I'd name the first variable `is_valid` rather than `success`, closer the parameter name in the header.

And then you can write this as:
```rs
match is_valid {
    0 => None,
    1 => Some(variant),
    _ => panic!("Invalid valid status {is_valid}"),
}
```

Reply:
that would make it misleading, as valid is a different boolean from success. valid represents whether the variant sent in is valid or not, whereas the return is whether the function succeeded at returning an iterator

Reply:
I think it's what they use in the header though?

Success is fine for me too, important is the `match` 😉